lib/nmatrix/numruby.rb

```diff
@@ -62,4 +62,28 @@ def self.hstack(objs)
     result
   end
 
+  def self.tri(N, M=None, k=0, nm_dtype=float)
+    if M is None:
+      M = N
+
+      m = greater_equal.outer(arange(N, nm_dtype=nm_int(0, N)),
+        arange(-k, M-k, nm_dtype=nm_int(-k, M - k)))
+
+        # Avoid making a copy if the requested type is already bool
+        m = m.stype(nm_dtype, copy=False)
+        return m
+      end
+
```

Comment:
I think that missing and `end`

Reply:
I think lines `72` to `76` are having extra indentation.

Reply:
@Uditgulati I have removed unwanted spaces please take a look